src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -0,0 +1,423 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Text;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal class DecisionDagBuilder
+    {
+        private readonly Conversions Conversions;
+        private readonly TypeSymbol BooleanType;
+        private readonly TypeSymbol ObjectType;
+
+        internal DecisionDagBuilder(CSharpCompilation compilation)
+        {
+            this.Conversions = compilation.Conversions;
+            this.BooleanType = compilation.GetSpecialType(SpecialType.System_Boolean);
+            this.ObjectType = compilation.GetSpecialType(SpecialType.System_Object);
```

Comment:
>compilation.GetSpecialType(SpecialType.System_Object) [](start = 30, length = 53)

Do we need to report a use-site error for this type?

Reply:
No, as it would have been reported on the input expression during normal binding.

---
In reply to: [156531608](https://github.com/dotnet/roslyn/pull/23209#discussion_r156531608) [](ancestors = 156531608)